packages/block-editor/src/utils/object.js

```diff
@@ -42,16 +42,16 @@ function cloneObject( object ) {
 }
 
 /**
- * Perform an immutable set.
- * Handles nullish initial values.
- * Clones all nested objects in the specified object.
+ * Immutably sets a value inside an object. Like `lodash#set`, but returning a
+ * new object. Treats nullish initial values as empty objects. Clones any
+ * nested objects.
  *
  * @param {Object}              object Object to set a value in.
  * @param {number|string|Array} path   Path in the object to modify.
  * @param {*}                   value  New value to set.
  * @return {Object} Cloned object with the new value set.
  */
-export function immutableSet( object, path, value ) {
+export function setImmutably( object, path, value ) {
```

Comment:
I just wanted to note that from my perspective, `immutableSet` wasn't a bad name, recognizable since it exists in other languages (Java AFAIK) and also there's Immutable.js and an `immutable-set` package. I've never seen `setImmutably` in code, rather I've seen `setImmutable`. I don't have very strong feelings about it though.

Reply:
To be honest, I did have a small issue with the name, at first I also didn't know what the function did until I looked at the tests. It felt more as an object creator. But I generally don't feel strongly about the naming things.

Reply:
Alright, kids! Then I guess I'll merge once it's green. :)